### ISSUE TITLE
Callback on scheduled notification shown (Android)

### ISSUE THREAD

**Issue by noinskit** [MaikuB/flutter_local_notifications#1699]
Now that the plugin is open to running Flutter code in the background, what about optionally firing a callback when a scheduled notification is *shown* to the user? I think it's only doable on Android, so it would probably be an optional callback in `AndroidInitializationSettings`, with similar restrictions as `onDidReceiveBackgroundNotificationResponse` in `initialize()`.

If the idea sounds reasonable, I might be able to prepare a merge request on my own.

**Comment by MaikuB** [comment]
Yep be open to a PR on this and if it's specific to Android then it should be specific to the Android implementaiton of the plugin. From my understanding, this is only possible on Android like you said as the callback on iOS/macOS   that takes place before a notification is shown only triggers when the app is in the foreground (see https://developer.apple.com/documentation/usernotifications/unusernotificationcenterdelegate/1649518-usernotificationcenter)

**Comment by pbayog00** [comment]
Are there any updates on this? I have not been able to find any PR related to this. 
Are we still not able to be aware of when a notification is shown to the user, regardless of it beeing tapped or not? Both for android and iOS

**Comment by Gaurav-CareMonitor** [comment]
@MaikuB Why is this not possible in iOS , since firebase also has a similar callback and it gets triggered in background in iOS

**Comment by MaikuB** [comment]
Firebase deals with push notifications not local notifications so you're not doing a direct comparison. If you want to know then you should look at the notification APIs to understand how they work and the capabilities they expose

**Comment by necropolina** [comment]
Is any work being done on this already? Would be great to have for android. Dunno if I could make time to implement it myself anytime soon since I'm unfamiliar with this codebase.

**Comment by Levi-Lesches** [comment]
For anyone interested, the relevant lines in Java are: 

https://github.com/MaikuB/flutter_local_notifications/blob/b5241cc61d6db2e41ccda6effb854a69613ffb66/flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java#L57-L62

To call a callback in Dart, you'll need to do something similar to [`ActionBroadcastReceiver.onReceive`](https://github.com/MaikuB/flutter_local_notifications/blob/b5241cc61d6db2e41ccda6effb854a69613ffb66/flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ActionBroadcastReceiver.java#L45)
